### ISSUE TITLE
rocket_codegen v0.3.9 AST mismatch failure on Rust nightly-2018-05-03

### ISSUE THREAD

**Issue by BartMassey** [SergioBenitez/Rocket#629]
`rocket_codegen` v0.3.9 from Cargo builds fine on Rust nightly-2018-04-28, but fails on Rust nightly-2018-05-03 due to AST mismatch.

```
error[E0609]: no field `name` on type `syntax::ast::Path`
  --> /home/bart/.cargo/registry/src/github.com-1ecc6299db9ec823/rocket_codegen-0.3.9/src/utils/meta_item_ext.rs:13:63
   |
13 |             MetaItemKind::NameValue(ref l) => Some((&mi.ident.name, l)),
   |                                                               ^^^^ unknown field
   |
   = note: available fields are: `span`, `segments`

error[E0609]: no field `name` on type `syntax::ast::Path`
   --> /home/bart/.cargo/registry/src/github.com-1ecc6299db9ec823/rocket_codegen-0.3.9/src/parser/route.rs:173:58
    |
173 |         if let Ok(method) = Method::from_str(&word.ident.name.as_str()) {
    |                                                          ^^^^ unknown field
    |
    = note: available fields are: `span`, `segments`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0609`.
error: Could not compile `rocket_codegen`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by SergioBenitez** [comment]
Fixed in v0.3.10. 🎉

**Comment by BartMassey** [comment]
Thank you!
